### PR TITLE
Versions updated & reorganization

### PR DESCRIPTION
**Pull request by iCesofT**
- Spring boot to 2.2.7.RELEASE
- Reorganization (dependencyManagement vs dependencies)

### CODE REVIEW

**Comment by martinalig** [comment]
Hi
Thanks for the inputs. 
One think: We do not plan to replace logback with log4j2. Please adapt your pullrequest.

**Comment by iCesofT** [comment]
I already changed my code to use logback (using spring-boot-starter-logging).

File: README.md [hunk 1]
```diff
@@ -1,30 +1,13 @@
+<!-- Issue templates and Contributing guides are taken from the fastlane repository and adjusted for our needs (https://github.com/fastlane/fastlane/tree/master/.github/ISSUE_TEMPLATE) and  -->
```
Comment (MichaelPFrey) [inline README.md:1]: The `and` at the end of the sentence looks like a stray/artifact. Or got something lost while copy-editing?
```suggestion
<!-- Issue templates and Contributing guides are taken from the fastlane repository (https://github.com/fastlane/fastlane/tree/master/.github/ISSUE_TEMPLATE) and adjusted for our needs  -->
```
From a readability perspective, I think would be better to place the URL directly to where the repository is mentioned - but that is personal preference.

File: dpppt-backend-sdk/dpppt-backend-sdk-data/src/main/environment/production/db/migration/hsqldb/V0_5__gaen_exposed_keys.sql [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/*
+ * Created by Ubique Innovation AG
+ * https://www.ubique.ch
+ * Copyright (c) 2020. All rights reserved.
```
Comment (MichaelPFrey) [inline dpppt-backend-sdk/dpppt-backend-sdk-data/src/main/environment/production/db/migration/hsqldb/V0_5__gaen_exposed_keys.sql:4]: Nitpick//Observation/Question: `All rights reserved.` Is this an outdated copyright headers?
#75 looks like it replaced the `All rights reserved.` with `MPL-2.0`
https://github.com/DP-3T/dp3t-sdk-backend/pull/75/files
_(This applies to other files of this PR, but I only make one comment - maybe there is a good reason for this header)_

File: dpppt-backend-sdk/dpppt-backend-sdk-ws/src/main/java/org/dpppt/backend/sdk/ws/security/KeyVault.java [hunk 3]
```diff
@@ -152,16 +152,23 @@ public static PrivateKey loadPrivateKeyFromJavaEncoding(String privatePart, Stri
...
-			return KeyFactory.getInstance(algorithm).generatePrivate(new PKCS8EncodedKeySpec(obj.getContent()));
+			result = KeyFactory.getInstance(algorithm).generatePrivate(new PKCS8EncodedKeySpec(obj.getContent()));
 		} catch (InvalidKeySpecException | NoSuchAlgorithmException | IOException e) {
 			logger.error("Exception loading private key from PEM", e);
-			return null;
+		} finally {
+			try {
+				readerPem.close();
+			} catch (IOException e) {
+					logger.error("Exception closing PEM reader: {}", e.getMessage(), e);
```
Comment (MichaelPFrey) [inline dpppt-backend-sdk/dpppt-backend-sdk-ws/src/main/java/org/dpppt/backend/sdk/ws/security/KeyVault.java:168]: nitpick: this looks like one tab to many.
(`readerPem` and `logger` should have the same intend, just looking at the brackets)
```suggestion
				logger.error("Exception closing PEM reader: {}", e.getMessage(), e);
```

**Comment by ineiti** [comment]
Hi - I'm an external contributor, so my comments do not reflect Ubique's team...

- the PR should be rebased against the latest `develop` branch
- I see a lot of changes in the documentation and different pieces of code - is this because it's not rebased?
- if there are still a lot of changes even after the rebase, I would propose to:
  - explain what you want to do by opening an issue, eventually more than one
  - create one PR per issue

**Comment by iCesofT** [comment]
I will create a new PR.

**Pull request closed without merging**